### PR TITLE
Handle client abort and server timeout separatly.

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -62,6 +78,7 @@ function pinoLogger (opts, stream) {
     res.on('error', onResFinished)
     // it's possible that browser aborts connection, or http-server because of timeout
     req.on('aborted', onReqAborted)
+    req.socket.on('timeout', onReqTimeout.bind(req))
```
Comment (ronag) [inline logger.js]: Not sure why `req.on('timeout', ...)` doesn't do the same thing...

File: test.js [hunk 2]
```diff
@@ -291,9 +291,8 @@ test('react on aborted event from clientside', function (t) {
   }, logger)
 
   dest.on('data', function (line) {
-    t.equal(line.msg, 'request errored', 'message is set')
-    t.equal(line.err.message, 'Aborted', 'error message is set')
-    t.equal(line.res.statusCode, 408, 'statusCode is 408')
+    t.equal(line.msg, 'request aborted', 'message is set')
+    // t.equal(line.err.message, 'Aborted', 'error message is set')
```
Comment (ronag) [inline test.js]: Need separate test for safe and non-safe methods.

File: logger.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ function pinoLogger (opts, stream) {
   loggingMiddleware.logger = logger
   return loggingMiddleware
 
-  function onResFinished (err) {
+  function onResFinished (err, msg = 'completed') {
```
Comment (mcollina) [inline logger.js]: `pino-http` supports node v4+, and this syntax is available only on node v6+. Moreover, it is slower than using `||` to set a default.

File: logger.js (hunk 1 above)
Comment (mcollina) [inline logger.js]: I think `'timeout'`  is not emitted if `setTimeout()`  is not called on the request or response.
Maybe is what we want, would you mind setting `setTimeout` explicitly?

`bind` is extremely slow on node v4 and v6, it should not be used.

Comment (ronag) [inline logger.js]: I don't think it's a good idea to override the timeout.

Comment (mcollina) [inline logger.js]: I agree, but that's why `req.on('timeout')` is not firing.

**Comment by ronag** [comment]
@mcollina Updated

**Comment by mcollina** [comment]
Released as v2.4.0.

**Pull request merged**